mongodb/controllers/getRestaurantById.js

```diff
@@ -1,43 +1,35 @@
-// const redisClient = require('redis').createClient;
-
-// const redis = redisClient(6379, 'localhost');
 const redis = require('redis');
 const { promisify } = require('util');
 const Promise = require('bluebird');
 
+const database = require('../models/restaurant.js');
+
 const client = redis.createClient();
 
 const getAsync = promisify(client.get).bind(client);
-const setAsync = promisify(client.set).bind(client);
-// const existsAsync = promisify(client.exists).bind(client);
-
+const setexAsync = promisify(client.setex).bind(client);
 
 client.on('error', (err) => {
   console.error(`Error ${err}`);
 });
 
-
-const database = require('../models/restaurant.js');
-
-const getRestaurantById = id => database.find({ place_id: id })
-  .then(result => result[0]);
-
-const getRestaurantByIdCached = (id) => {
+const getRestaurantById = (id) => {
   return new Promise((resolve, reject) => {
     client.exists(id, (err, reply) => {
+      if (err) reject(err);
       if (reply === 1) {
         getAsync(id).then((res) => {
           resolve(JSON.parse(res)[0]);
         });
       } else {
         database.find({ place_id: id })
           .then((result) => {
-            setAsync(id, JSON.stringify(result));
+            setexAsync(id, 3600, JSON.stringify(result));
```

Comment:
@annisakaraca -Yes, I set the cache to have items expire after an hour. My initial thought was an hour just because it seemed roughly aligned with the period that users might peak, for example, an hour or so period right before lunch or dinner. Just my first thought though. It would be interesting to consider the optimal cache expiry and test. Let me know if you have any suggestions!